wagtail/admin/forms/collections.py

```diff
@@ -19,6 +20,18 @@
 
 
 class CollectionViewRestrictionForm(BaseViewRestrictionForm):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not getattr(settings, "WAGTAIL_ALLOW_SHARED_PASSWORD_COLLECTION", True):
```

Comment:
Issue: I think this change would be better made on the `BaseViewRestrictionForm`, as it'll then apply everywhere. Given that's where the view restrictions are defined in the first place.

Reply:
I made changes in `PageViewRestriction` and `CollectionViewRestriction` because I believed that if someone wanted to set only `WAGTAIL_ALLOW_SHARED_PASSWORD_PAGE` or `WAGTAIL_ALLOW_SHARED_PASSWORD_COLLECTION`, putting the change in `BaseViewRestrictionForm` would remove shared password field both in pages and collections. Am I mistaken?

Reply:
Ah, yes that's very true. What's the thinking behind having these separately controllable? I suspect someone is most likely to want to disable it outright, rather than separately like that.